python/foglamp/plugins/south/systeminfo/systeminfo.py

```diff
@@ -0,0 +1,183 @@
+        'type': 'string',
+        'default': 'systeminfo'
+    },
+    'assetCode': {
+        'description': 'Asset Code',
+        'type': 'string',
+        'default': "system"
+    },
+    'sleepInterval': {
+        'description': 'Sleep Interval, in seconds, between two System info gathering',
+        'type': 'integer',
+        'default': "30"
+    }
+}
+_LOGGER = logger.setup(__name__, level=20)
+
+
+def plugin_info():
+    """ Returns information about the plugin.
+    Args:
+    Returns:
+        dict: plugin information
+    Raises:
+    """
+
+    return {
+        'name': 'System Info plugin',
+        'version': '1.0',
+        'mode': 'async',
+        'type': 'south',
+        'interface': '1.0',
+        'config': _DEFAULT_CONFIG
+    }
+
+
+def plugin_init(config):
+    """ Initialise the plugin.
+    Args:
+        config: JSON configuration document for the South device configuration category
+    Returns:
+        data: JSON object to be used in future calls to the plugin
+    Raises:
+    """
+    data = copy.deepcopy(config)
+    return data
+
+
+def plugin_start(handle):
+    """ Extracts data from the system info and returns it in a JSON document as a Python dict.
+    Available for async mode only.
+
+    Args:
+        handle: handle returned by the plugin initialisation call
+    Returns:
+        a system info reading in a JSON document, as a Python dict, if it is available
+        None - If no reading is available
+    Raises:
+        TimeoutError
+    """
+    def get_system_info():
+        # Details of machine resources, memory size, amount of available memory, storage size and amount of free storage
+        total, used, free = shutil.disk_usage("/")
+        memory = subprocess.Popen('free -h', shell=True, stdout=subprocess.PIPE).stdout.readlines()[1].split()[1:]
+        data = {
+            "platform": sys.platform,
+            "totalMemory": memory[0].decode(),
+            "usedMemory": memory[1].decode(),
+            "freeMemory": memory[2].decode(),
+            "totalDiskSpace_MB": int(total / (1024 * 1024)),
+            "usedDiskSpace_MB": int(used / (1024 * 1024)),
+            "freeDiskSpace_MB": int(free / (1024 * 1024)),
+        }
+
+        # A PS listing of al the python applications running on the machine
+        a = subprocess.Popen(
+            'ps -eaf', shell=True, stdout=subprocess.PIPE).stdout.readlines()[:-2]  # remove ps command
+        c = [b.decode() for b in a]  # Since "a" contains return value in bytes, convert it to string
+        data.update({
+            "numberOProcessesRunning": c
+        })
+
+        # Get Network and other info
+        data.update({
+            "numberOfNetworkPacketsReceived" : None,
+            "numberOfNetworkPacketsSent": None,
+            "cpuUsage": None,
+            "numberOfPagingAndSwappingEvents": None,
+            "diskTraffic": None
+        })
+
+        return data
+
+    async def save_data():
```

Comment:
we are using >=1.3 now, so utils.local_timestamp()
again, not sure if we need this as it may be poll plugin